config/network/config.go

```diff
@@ -1,26 +1,35 @@
 package network
 
 import (
+	"slices"
+
 	"github.com/UpCloudLtd/provider-upcloud/config/groupversion"
 	"github.com/crossplane/upjet/pkg/config"
 )
 
-// Resources is a list of all supported network resources.
-var Resources = []string{
-	"upcloud_network",
+// SDKResources is a list of all supported network resources implemented with Terraform legacy SDKv2.
+var SDKResources = []string{
 	"upcloud_router",
 }
 
+// PluginFrameworkResources is a list of all supported network resources implemented with Terraform Plugin Framework.
+var PluginFrameworkResources = []string{
+	"upcloud_network",
+}
+
+// AllResources is a list of all supported network resources.
+var AllResources = slices.Concat(SDKResources, PluginFrameworkResources)
+
 // Configure configures the network resources.
 func Configure(p *config.Provider) {
-	groupversion.Configure(Resources, p, "network", "v1alpha1")
+	groupversion.Configure(AllResources, p, "network", "v1alpha1")
 
 	p.AddResourceConfigurator("upcloud_network", func(r *config.Resource) {
 		r.ExternalName = config.IdentifierFromProvider
 
-		r.References["router"] = config.Reference{
-			Type: "Router",
-		}
+		// r.References["router"] = config.Reference{
+		// 	Type: "Router",
+		// }
```

Comment:
Commented out temporarily, since this resource doesn't work correctly and I wanted to see if it has anything to do with references.